inc/class-component.php

```diff
@@ -86,21 +86,52 @@ public static function get_component_list(): array {
 		}
 
 		// Scan the components directory to build a list of components.
-		$slugs = array_filter(
-			scandir( $components_dir ),
-			function ( $dir ) use ( $components_dir ) {
-				return '.' !== $dir
-					&& '..' !== $dir
-					&& is_dir( sprintf( '%s/%s', $components_dir, $dir ) );
-			}
-		);
+		$slugs = self::get_component_slugs_recursively_from_dir( $components_dir );
+
 		foreach ( $slugs as $slug ) {
 			$components[] = new self( $slug );
 		}
 
 		return $components;
 	}
 
+	/**
+	 * Recursively get slugs to instantiate components from folders/files in a directory.
+	 *
+	 * @param string $components_dir The dir containing components.
+	 * @return array
+	 */
+	private static function get_component_slugs_recursively_from_dir( string $components_dir ): array {
+		$slugs     = [];
+		$directory = new \RecursiveDirectoryIterator( $components_dir, \FilesystemIterator::FOLLOW_SYMLINKS );
```

Comment:
Can `use RecursiveDirectoryIterator` up top and then not need the fully qualified path here, etc for other classes

Reply:
Good catch